pkgs/development/python-modules/pylette/default.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  fetchFromGitHub,
+  poetry-core,
+  scikit-learn,
+  typer,
+  requests,
+  pillow,
+  numpy_1,
+  scipy,
+  pytestCheckHook,
+  opencv-python,
+  opencv4,
+  requests-mock,
+}:
+buildPythonPackage rec {
+  pname = "pylette";
+  version = "4.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "qTipTip";
+    repo = "Pylette";
+    tag = version;
+    hash = "sha256-i8+QQpYoRfgoV9Nm/FvXSJV+TEvmaaPsPIeG+PU838Q=";
+  };
+
+  build-system = [
+    poetry-core
+  ];
+
+  dependencies = [
+    (scikit-learn.overridePythonAttrs (old: {
+      dependencies = (builtins.filter (x: x.pname != "numpy" && x.pname != "scipy") old.dependencies) ++ [
+        numpy_1
+        (scipy.overridePythonAttrs (old: {
+          dependencies = [ numpy_1 ];
+        }))
+      ];
+    }))
+    (pillow.overridePythonAttrs (old: rec {
+      version = "10.4.0";
+      src = fetchPypi {
+        inherit version;
+        pname = "pillow";
+        hash = "sha256-Fmwc1NJDCbMNYfefSpEUt7IxPXRQkSJ3hV/139fNSgY=";
+      };
+    }))
+    requests
+    typer
+    numpy_1
```

Comment:
We can't accept the propagation of a versioned package since it causes a PYTHONPATH conflict.

see more information, https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

Reply:
Yeah, that's a bummer... I fixed it. It seems like with a simple `substituteInPlace` on problematic packages solved this. I've tested the bin & the lib, everything seems fine now.